added small team responsibilities, defined the role of the team lead and cleared up how PMs work with engineers

lharries opened the pull request:
We've are shifting from teams being based around geographies to small teams owning areas of the app. This has resulted in some confusion around:
1. what are the responsibilities of these teams
2. what actions should they be doing
3. which parts do the team leads need to do
4. how do the product managers fit into this

This is a first draft of an attempt to answer these questions.

lharries commented:
@annikaschmid @timgl @benjackwhite @EDsCODE @mariusandra @jamesefhawkins @yakkomajuri this is based on conversations we've had. Would be great to get your input on this

benjackwhite commented on contents/handbook/people/team-structure/app-small-teams.md:
```diff
@@ -0,0 +1,41 @@
+
+## What is the role of the team lead?
+
+The team lead should led the team to make sure the above happens. They should get the team fully involved rather that doing everything themselves.
```
🤔 "team lead should lead the team" ?

```suggestion
The team lead is overall responsible for ensuring the above happens. They should focus on enabling the team to solve these tasks together rather than trying to do it all themself.
```

macobo commented on contents/handbook/people/team-structure/app-small-teams.md:
```diff
@@ -0,0 +1,41 @@
+
+With our engineering-led culture, the engineers on the app small team are responsible for the product. 
+
+We have a small number of product managers who support the app small teams in achieving their goals. This includes a wide range of activities such as helping with prioritization, setting up dashboards, competitors analysis, speaking to customers etc.
```
Of these, setting up dashboards seems like the odd one out for me - given the teams priorities are clear, anyone on the team should be empowered to set up and maintain metrics.

pauldambra replied:
>  setting up dashboards seems like the odd one out for me - given the teams priorities are clear, anyone on the team should be empowered to set up and maintain metrics.

Yep, feels like, as with prioritization, product folk should be the local experts enabling everyone to do it.

lharries replied:
I agree with this "anyone on the team should be empowered to set up and maintain metrics" @macobo and was what I meant with that sentence. Any ideas how to frame it better?

macobo commented on contents/handbook/people/team-structure/app-small-teams.md:
```diff
@@ -0,0 +1,41 @@
+  1. Create good quarterly OKRs
+  2. Create an ambitious vision for their area
+- During the quarter:
+  1. Maintain a prioritized roadmap to help them achieve their objectives
```
What problem does having a roadmap for each team solve and who will be it's main consumers?

(I wrote in-depth about why I'm not a fan of how roadmap-driven development usually falls out here: https://github.com/PostHog/posthog/issues/11749#issuecomment-1253302481)

pauldambra replied:
Does roadmap here really mean backlog? But as Karl says "roadmap" "backlog" and similar are overloaded and can reduce clarity

If we rewrote as `frequently prioritise upcoming work to remain focussed on their objectives`... does that help guide what we do without restricting how we do it?

lharries replied:
What Annika and Paul are saying with "frequently prioritise upcoming work to remain focussed on their objectives" is more what I'm thinking. It's not about having a strict list of stuff they must deliver, but more a space for them to keep track of potential ideas for the future sprints.

lharries replied:
I'll update this now

lharries replied:
What do you think about there being a more formal requirement that this is saved as a stack ranked list that is shared internally? We don't need to be prescriptive of the format or location but I know it would help me and likely others if there was a linked list written down somewhere?

lharries replied:
So the line would be "Frequently prioritize upcoming work to help them achieve their objectives (the format does not matter but the link to this should be shared)"

pauldambra replied:
That's fine by me... 

We need to keep each other honest so it doesn't become too large a list or too process-y to keep it up-to-date

("the planning is essential but the plans are worthless" - Dwight D. Eisenhower)

mariusandra replied:
I'm not sure what line should end up in this doc, but at least in Team Product Analytics, we're dumping every relevant issue that comes our way onto [the project's board here](https://github.com/orgs/PostHog/projects/59), and at least each week coming back to see what urgent bugs are there to distract us from our OKRs :).

pauldambra commented on contents/handbook/people/team-structure/app-small-teams.md:
```diff
@@ -0,0 +1,41 @@
+
+The team lead is overall responsible for ensuring the above happens. They should focus on enabling the team to solve these tasks together rather than trying to do it all themself.
+
+## What do the product managers do?
```
Do we need a `## What do the designers do?` section?

lharries replied:
Good point. This is a bigger topic of discussion and is still being clarified

@corywatilo if you feel confident writing a section here go for it otherwise we can add a link to https://github.com/PostHog/posthog.com/pull/4003/files after it's published

pauldambra commented on contents/handbook/people/team-structure/app-small-teams.md:
```diff
@@ -0,0 +1,41 @@
+hideAnchor: false
+---
+
+App Small Teams are particular types of Small Teams that own an area of the product.
```
Is "App" too overloaded a term

I've seen these called "product teams" or ["stream-aligned teams"](https://teamtopologies.com/key-concepts)

lharries replied:
I was choosing between "App team", "Product team" and "Feature team". On reflection I also think "Product team" makes the most sense and makes it clear the team is responsible for their product area holistically

pauldambra commented on contents/handbook/people/team-structure/app-small-teams.md:
```diff
@@ -0,0 +1,41 @@
+
+App Small Teams are particular types of Small Teams that own an area of the product.
+
+## What does owning an area of the product mean?
```
Also makes me want to ask "what are the areas of the product?"

lharries replied:
not sure I understand, would you mind rephrasing?

pauldambra replied:
I suppose I'm asking... do we have a list of the products/product teams?

* session recordings
* product analytics
* flags and experimentation
* ingestion
* platform

??

Is every team a product team?

pauldambra reviewed (commented):
Few questions and comments but overall better that we have this 💪

annikaschmid commented on contents/handbook/people/team-structure/app-small-teams.md:
```diff
@@ -0,0 +1,41 @@
+  3. Monitor relevant metrics including those covering Usage, Quality and Revenue
+  4. Triage and fix related bugs
+  5. Assist the support hero in answering related questions
+  6. Collaborate with marketing
```
I would also add a cheeky point here: "Become power users of their app area and use PostHog in their product development process" 
I know that we are doing this already so some extent, but I would love to specify this as part of this ways of working proposal

jamesefhawkins commented on contents/handbook/people/team-structure/app-small-teams.md:
```diff
@@ -0,0 +1,41 @@
+---
+title: Product Small Teams
```
feedback: this title confused me! (ie we treat website and docs like a product too / customer success also should be treated like a product)

I'd probably just work this content into the existing page that explains small teams: https://posthog.com/handbook/people/team-structure/team-structure

I generally think though the content here is valuable and clarifies a bunch of things!

mariusandra replied:
+1 to that. I'd say we're "small teams", and making a distinction "product small teams", "infra small teams", "marketing small teams", etc loses the "everyone's a full stack product minded engineer" mindset. Also, infra, marketing, customer success, etc are all part of our product.

lharries replied:
Good suggestions and I agree with the reasoning. I'll refactor into the team-structure page

lharries commented:
@jamesefhawkins and @timgl I've addressed the above comments.

Let me know what you think and approve it if you think we are ready to merge - or let me know if there are other parts you think we should address.

timgl reviewed (approved):
Sorry, hadn't had time to look at this yet. All the discussions above were great and have led to the right outcome here, so I'm happy to merge!

jamesefhawkins commented:
I'm going to merge. Thank you everyone for contributing.

@lharries I made a couple of tweaks above, just to simplify:

- each small team doesn't have to have a mission/vision and metrics (given just OKRs are new already - and there is a lot of overlap here)
- I have _deliberately_ left in they should have a customer. I think this is something we are not doing well at the moment. We have an ideal customer profile, but it's the same for everyone and could be much more precise by small team. Might be something you want to think about.

I'm merging as this is better than what we had, feel free to open future PRs to address the above as you see fit

kappa90 commented on contents/handbook/people/team-structure/why-small-teams.md:
```diff
@@ -74,7 +112,9 @@ It is at the discretion of the _manager_ of that person if they can move.
-In certain cases, but not everywhere. This will clarify where people will work. In fact, it'll make sure we keep the scrappy fun side of working here as we get bigger. A team doesn't _have_ to be six people.
+In general, no - it's surprisingly great how much just 2-6 people can get done.
+
+If more mature product areas cannot cope with the workload, smallt eams will clarify where we need to hire too. In fact, it'll make sure we keep the scrappy fun side of working here as we get bigger. A team doesn't _have_ to be six people.
```
typo! `smallt eams`
